assemblies/assembly-optimizer-migration-8.adoc

```diff
@@ -0,0 +1,1685 @@
+[source,xml]
+----
+  ...
+  import org.optaplanner.core.impl.score.director.incremental.AbstractIncrementalScoreCalculator;
+  ...
+
+  public class CloudBalancingIncrementalScoreCalculator extends AbstractIncrementalScoreCalculator<CloudBalance> {
+    ...
+  }
+----
+
+After in `*IncrementalScoreCalculator.java*`:
+
+[source,xml]
+----
+  ...
+  import org.optaplanner.core.api.score.calculator.IncrementalScoreCalculator;
+  ...
+
+  public class CloudBalancingIncrementalScoreCalculator implements IncrementalScoreCalculator<CloudBalance, HardSoftScore> {
+    ...
+  }
+----
+
+Before in `*ConstraintMatchAwareIncrementalScoreCalculator.java*`:
+
+[source,xml]
+----
+  ...
+  import org.optaplanner.core.impl.score.director.incremental.AbstractIncrementalScoreCalculator;
+  import org.optaplanner.core.impl.score.director.incremental.ConstraintMatchAwareIncrementalScoreCalculator;
+  ...
+
+  public class CheapTimeConstraintMatchAwareIncrementalScoreCalculator
+        extends AbstractIncrementalScoreCalculator<CheapTimeSolution>
+        implements ConstraintMatchAwareIncrementalScoreCalculator<CheapTimeSolution> {
+    ...
+  }
+----
+
+After in `*ConstraintMatchAwareIncrementalScoreCalculator.java*`:
+
+[source,xml]
+----
+  ...
+  import org.optaplanner.core.api.score.calculator.ConstraintMatchAwareIncrementalScoreCalculator;
+  ...
+
+  public class CheapTimeConstraintMatchAwareIncrementalScoreCalculator
+        implements ConstraintMatchAwareIncrementalScoreCalculator<CheapTimeSolution, HardMediumSoftLongScore> {
+    ...
+  }
+----
+
+[discrete]
+=== `PlannerBenchmarkFactory`: `createFromSolverFactory()` removed
+*Major*, *Public API*
+
+The method `PlannerBenchmarkFactory.createFromSolverFactory()` has long been deprecated in favor of
+`PlannerBenchmarkFactory.createFromSolverConfigXmlResource(String)`.
+It has now been removed.
+
+Before in `*.java`:
+
+[source,java]
+----
+SolverFactory<CloudBalance> solverFactory = SolverFactory.createFromXmlResource(
+        ".../cloudBalancingSolverConfig.xml");
+PlannerBenchmarkFactory benchmarkFactory = PlannerBenchmarkFactory.createFromSolverFactory(solverFactory);
+----
+
+After in `*.java`:
+
+[source,java]
+----
+PlannerBenchmarkFactory benchmarkFactory = PlannerBenchmarkFactory.createFromSolverConfigXmlResource(
+        ".../cloudBalancingSolverConfig.xml");
+----
+
+If you programmatically adjust the solver configuration, you can use `PlannerBenchmarkConfig.createFromSolverConfig(SolverConfig)`
+and then `PlannerBenchmarkFactory.create(PlannerBenchmarkConfig)` instead.
+
+[discrete]
+=== `PlannerBenchmarkFactory`: `getPlannerBenchmarkConfig()` removed
+*Minor*, *Public API*
+
+The method `PlannerBenchmarkFactory.getPlannerBenchmarkConfig()` has long been deprecated in favor of
+`PlannerBenchmarkFactory.create(PlannerBenchmarkConfig)`.
+A `PlannerBenchmarkConfig` is now instantiated before a `PlannerBenchmarkFactory` is instantiated, which is more natural.
+The old was has now been removed.
```

Comment:
If you could improve this sentence that would be great. I'll backport the fix to upstream.